tools/eslint-plugin/dist/rules/require-version-in-deprecation.js

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+//Custom rule that enforces version and description for @deprecated
+
+"use strict";
+
+const { getParserServices } = require("./utils/parser");
+const ts = require("typescript");
+
+module.exports = {
+  create(context) {
+
+    const parserServices = getParserServices(context);
+
+    function checkJsDoc(declaration, node) {
+      let tags = ts.getJSDocTags(declaration);
+
+      if (!tags || tags.length == 0) {
+        return;
+      }
+      for (let tag of tags) {
+        if (tag.tagName.escapedText == "deprecated") {
+
+          if (tag.comment == undefined) {
+            context.report(node, "@deprecated version and description missing");
+          }
+          else if (!(/\d+(\.(\d|x))+.{5,}/.test(String(tag.comment)))) {
+            context.report(node, "@deprecated missing version or description.");
+          }
+        }
+      }
+    }
+
+    return {
+      ExportNamedDeclaration(node) {
```

Comment:
The `Program` node has a property called `comments` that contains all the comments in a file. I think that will be way easier to handle than going through all the possible exported node types.

Reply:
demo here: https://astexplorer.net/#/gist/6e3ae05c11e0de3807b9f05e5d8167f6/700b2ec8ac49c5ca34f4e50607d51f3d5a0daf10